tests/FileStorageTest.php

```diff
@@ -14,25 +14,264 @@
+    }
+
+    public function testCrossProviderStorageDefaultS3Abs()
+    {
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $storage = $this->client->createS3FileStorage(self::DEFAULT_S3_OPTIONS);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage(sprintf('File storage with id "%d" and provider "azure" was not found.', $storage['id']));
+        $this->client->setAbsFileStorageAsDefault($storage['id']);
+    }
+
+    public function testCrossProviderStorageCredentialsRotateS3Abs()
+    {
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $storage = $this->client->createS3FileStorage(self::DEFAULT_S3_OPTIONS);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage(sprintf('File storage with id "%d" and provider "azure" was not found.', $storage['id']));
+        $this->client->rotateAbsFileStorageCredentials($storage['id'], self::ROTATE_ABS_OPTIONS);
+    }
+
+    public function testCrossProviderStorageCredentialsRotateAbsS3()
+    {
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $storage = $this->client->createAbsFileStorage(self::DEFAULT_ABS_OPTIONS);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage(sprintf('File storage with id "%d" and provider "aws" was not found.', $storage['id']));
+        $this->client->rotateS3FileStorageCredentials($storage['id'], self::ROTATE_S3_OPTIONS);
+    }
+
+    public function testCreateS3StorageWithoutRequiredParam()
+    {
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage("Missing parameters in request: 
+[awsKey] : This field is missing.
+[awsSecret] : This field is missing.
+[filesBucket] : This field is missing.
+[owner] : This field is missing.
+[region] : This field is missing.");
+        $this->client->createS3FileStorage([]);
+    }
+
+    public function testCreateAbsStorageWithoutRequiredParam()
+    {
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage("Missing parameters in request: 
+[accountName] : This field is missing.
+[accountKey] : This field is missing.
+[containerName] : This field is missing.
+[owner] : This field is missing.");
+        $this->client->createAbsFileStorage([]);
+    }
+
+    public function testRotateS3CredentialsWithoutRequiredParams()
+    {
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $storage = $this->client->createS3FileStorage(self::DEFAULT_S3_OPTIONS);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage("Missing parameters in request: 
```

Comment:
Tenhle endpoint je úplně nový, jestli dobře koukám do https://keboolamanagementapi.docs.apiary.io/#reference/super-file-storage-management/file-storage-collection/list-storages

A není mi úplně jasné, proč potřebuju ty původní credentials (které ideálně nemám). Proto jsou ty credentials uložené v DB, ne? A mě stačí, že mám superuser manage api key

Reply:
Doplnil jsem apiary. 

To jsou nový credentials, co v tomhle testu "chybí".